hnix-store-core/tests/NarFormat.hs

```diff
@@ -361,12 +363,16 @@ packThenExtract testName setup =
         pure ()
 
 -- | Count file descriptors owned by the current process
-countProcessFiles :: IO Int
+countProcessFiles :: IO (Maybe Int)
 countProcessFiles = do
   pid <- Unix.getProcessID
-  let fdDir = "/proc/" <> show pid <> "/fd"
-  fds  <- P.readProcess "ls" [fdDir] ""
-  pure $ length $ words fds
+  hasProc <- doesDirectoryExist "/proc"
+  if not hasProc
+    then pure Nothing
+    else do
+      let fdDir = "/proc/" <> show pid <> "/fd"
+      fds  <- P.readProcess "ls" [fdDir] ""
+      pure $ pure $ length $ words fds
```

Comment:
`pure` can be factored-out, `Nothing` is `mempty`. It would be seen that what return in previous structure with the internal value computed here.

It may not look as important at the moment, but over time.